src/libsync/networkjobs/resources.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "account.h"
+#include "networkjobs.h"
+
+#include <QIcon>
+#include <QLoggingCategory>
+
+namespace OCC {
+
+class ResourcesCache;
+
+/**
+ * This job automatically downloads all available data from the server and stores it in a temporary cache directory on the disk.
+ * For convenience, a couple of conversion functions are available to convert the binary data to common Qt classes such as QIcon.
+ */
+class OCSYNC_EXPORT ResourceJob : public SimpleNetworkJob
+{
+public:
+    void finished() override;
+
+    QIcon asIcon() const;
+
+    const QByteArray &data() const;
+
+protected:
+    explicit ResourceJob(const ResourcesCache *cache, const QUrl &rootUrl, const QString &path, QObject *parent = nullptr);
+    ;
+
+private:
+    const ResourcesCache *_cache;
+    QByteArray _data;
+    QString _cacheKey;
+
+    friend class ResourcesCache;
+};
+
+
+class OCSYNC_EXPORT ResourcesCache : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit ResourcesCache(Account *account);
+
+    ResourceJob *makeGetJob(const QUrl &rootUrl, const QString &path, QObject *parent = nullptr) const;
```

Comment:
I guess we should require a parent? 
Or the account should be the parent, worst outcome there is that they are not canceled before completion.
Related: https://github.com/owncloud/client/issues/10622

Reply:
We can remove the parameter's default.